pkg/handler/handler.go

```diff
@@ -70,18 +70,35 @@ func WebhookHandler(w http.ResponseWriter, r *http.Request, reconciler gitopscon
 				Items:    make([]gitopsv1alpha1.GitOpsConfig, len(list.Items)),
 			}
 
+			// The targetList.Items is initialized with an empty GitOpsConfig at index 0
+			// We are dropping this item to ensure we do not try to start a reconsiliation job for an "empty" GitOpsConfig resource
+			targetList.Items = targetList.Items[1:]
+
 			for _, instance := range list.Items {
-				// if does not have the webhook trigger continue
 				if !gitopsconfig.ContainsTrigger(&instance, "Webhook") {
+					log.Info("skip instance without webhook trigger", "instance_name", &instance.Name)
 					continue
 				}
-				// if the repo URL do not correspond continue
+
+				log.Info("comparing instance and event metadata", "event_name", e.Repo.GetFullName(), "event_ref", e.GetRef(),
+					"template_uri", instance.Spec.TemplateSource.URI, "template_ref", instance.Spec.TemplateSource.Ref,
+					"parameter_uri", instance.Spec.ParameterSource.URI, "parameter_ref", instance.Spec.ParameterSource.Ref)
+
 				if !repoURLAndRefMatch(&instance, e) {
+					log.Info("skip instance without matching repo url or git ref of the event", "instance_name", &instance.Name)
 					continue
 				}
+
+				log.Info("found matching instance", "instance_name", &instance.Name)
```

Comment:
Do we really need to pass a pointer here?

Reply:
@Smiley73 the pointer/reference is not needed, I have updated the logs in pkg/handler to use the instance name value.